### ISSUE TITLE
Cannot install ndless (3.6 CX)

### ISSUE THREAD

**Issue by Candunc** [nspire-emus/firebird#16]
I know this is an absolutely stupid request, but still. I was going to use it for the wiki.
ANYWAYS

When I execute ndless_installer.tns (OS 3.6 CX), the program crashes. I'd assume this is because ndless uses some sort of stack overflow in order to execute the hack, but the emulator probably doesn't take kindly to this.

```
Process:               firebird [1199]
Path:                  /Applications/firebird.app/Contents/MacOS/firebird
Identifier:            com.yourcompany.firebird
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           firebird [1199]
User ID:               501

Date/Time:             2015-09-20 17:35:17.528 -0600
OS Version:            Mac OS X 10.10.5 (14F27)
Report Version:        11
Anonymous UUID:        A0D40770-A6D3-E04A-BA5B-919F79F4085F

Sleep/Wake UUID:       D8C055FF-E0E7-44E4-BF07-91ACA364BCFA

Time Awake Since Boot: 11000 seconds
Time Since Wake:       6200 seconds

Crashed Thread:        4  EmuThread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000

VM Regions Near 0:
--> 
    __TEXT                 0000000100000000-00000001000a6000 [  664K] r-x/rwx SM=COW  /Applications/firebird.app/Contents/MacOS/firebird
```

**Comment by adriweb** [comment]
Weirdly enough, that has always worked, in fact it definitely should, testing Ndless apps on the emu is particularly important :P
Nevertheless, I can confirm I get the same crash on 10.11 with Ndless 3.6 on a CX CAS flash.

Works fine on 3.9.1 though...

**Comment by Candunc** [comment]
Ok, I'll try with 3.9.1. I tried with the exact same image with nspire_emu, and it installed fine on Windows. This might be a Mac Specific problem, I'll get back to you in a bit.

**Comment by Legimet** [comment]
I remember this used to happen in nspire_emu when I compiled it myself using 
the MinGW cross-compiler. I don't know if it was ever fixed.

**Comment by Candunc** [comment]
Hmm, 3.9.1 works perfectly. Seems to be an issue with 3.6 specifically. I'll add that to the wiki, thanks!

**Comment by adriweb** [comment]
Let's keep it open, it's... still a bug :|

**Comment by Vogtinator** [comment]
I believe fixed this issue in the past, it was related to the debugger dock.
The ndless_installer_3.6.tns causes a fault, so the debugger is entered if enabled (Settings->Emulation->Debug on: Warning) and it tries to raise the debugger dock. In tab mode, however, dockDebugger is NULL, so it crahed.
I'll have a look at this, should be easy to fix. In the meanwhile, can someone post a stacktrace of the crash, preferably with debug symbols (line numbers)?

**Comment by adriweb** [comment]
That still happened for me in Docks mode, and I also couldn't get a backtrace from within Qt's debugger, the guilty thread was gone...
I may retry with gdb.

**Comment by Candunc** [comment]
Adding onto this, I got a similar crash when launching a ndless application. I tried to make a debug build, however it crashes (Probably similar crash to earlier, need to disable docks).

```
Process:               firebird [475]
Path:                  /Applications/firebird.app/Contents/MacOS/firebird
Identifier:            com.yourcompany.firebird
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           firebird [475]
User ID:               501

Date/Time:             2015-09-21 10:15:02.256 -0600
OS Version:            Mac OS X 10.10.5 (14F27)
Report Version:        11
Anonymous UUID:        A0D40770-A6D3-E04A-BA5B-919F79F4085F


Time Awake Since Boot: 1200 seconds

Crashed Thread:        4  EmuThread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000

VM Regions Near 0:
--> 
    __TEXT                 0000000100000000-00000001000a6000 [  664K] r-x/rwx SM=COW  /Applications/firebird.app/Contents/MacOS/firebird
```

**Comment by Vogtinator** [comment]
The important part, the stack trace, is missing from your reports, without them it's hard to debug.

**Comment by Vogtinator** [comment]
Tried it in both docks and tabs mode, no crash with 3.6, even the debugger tab/dock raise worked correctly.

**Comment by adriweb** [comment]
Yeah, apparently it works fine on Linux and Windows, just not on Mac. 

Don't know about mobile...



Thanks,
Adrien Bertrand

**Comment by Vogtinator** [comment]
I've got a stacktrace, it's sadly easily reproducable...
Likely just another bug in Qt 5.5, I'll download 5.4 and test it.
![firebird smw1](https://cloud.githubusercontent.com/assets/1622084/10001116/787e2d9e-60a0-11e5-9702-585bc71abbed.png)

**Comment by adriweb** [comment]
This might help (pausing the Firebird Process when it's in its debugger during the data abort):
```
(lldb) bt
* thread #8: tid = 0x92cc6e, 0x00007fff8e433222 libsystem_kernel.dylib`__select + 10, name = 'EmuThread'
  * frame #0: 0x00007fff8e433222 libsystem_kernel.dylib`__select + 10
    frame #1: 0x0000000101975e04 QtCore`qt_safe_select(int, fd_set*, fd_set*, fd_set*, timespec const*) + 596
    frame #2: 0x0000000101976cd0 QtCore`QEventDispatcherUNIXPrivate::doSelect(QFlags<QEventLoop::ProcessEventsFlag>, timespec*) + 672
    frame #3: 0x0000000101977d9a QtCore`QEventDispatcherUNIX::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 234
    frame #4: 0x00000001019248cd QtCore`QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 381
    frame #5: 0x000000010002b9c6 firebird`::gui_debug_prompt() + 150 at emuthread.cpp:69
    frame #6: 0x000000010005a1da firebird`native_debugger() + 202 at debug.cpp:559
    frame #7: 0x000000010005a0e7 firebird`::debugger(reason=DBG_EXCEPTION, addr=0) + 71 at debug.cpp:742
    frame #8: 0x0000000100043617 firebird`warn(fmt="Data abort: address=%08x status=%02x instruction at %08x\n") + 439 at emu.c:69
    frame #9: 0x0000000100034411 firebird`data_abort(mva=16843021, status='\x05') + 49 at cpu.cpp:102
    frame #10: 0x0000000100052545 firebird`mmu_translate(addr=16843021, writing=true, fault=(firebird`data_abort at cpu.cpp:100), s_status=0x0000000000000000) + 245 at mmu.c:81
    frame #11: 0x0000000100052b9f firebird`addr_cache_miss(virt=16843021, writing=true, fault=(firebird`data_abort at cpu.cpp:100)) + 47 at mmu.c:230
    frame #12: 0x000000010000866e firebird`write_word(addr=16843021, value=286312500) + 110 at asmcode.c:149
    frame #13: 0x000000013225e00a
    frame #14: 0x00000001000340af firebird`::cpu_arm_loop() + 287 at cpu.cpp:41
    frame #15: 0x0000000100044001 firebird`emulate(port_gdb=3333, port_rdbg=3334) + 1617 at emu.c:312
    frame #16: 0x000000010002c1bd firebird`EmuThread::run(this=0x00007fff5fbff5e8) + 589 at emuthread.cpp:142
    frame #17: 0x000000010177cb53 QtCore`QThreadPrivate::start(void*) + 339
    frame #18: 0x00007fff98f079b1 libsystem_pthread.dylib`_pthread_body + 131
    frame #19: 0x00007fff98f0792e libsystem_pthread.dylib`_pthread_start + 168
    frame #20: 0x00007fff98f05385 libsystem_pthread.dylib`thread_start + 13
```
Some registers:
```
rax	unsigned long	0x000000000200005d	0x000000000200005d
rbx	unsigned long	0x0000000000000000	0x0000000000000000
rcx	unsigned long	0x0000700000322468	0x0000700000322468
rdx	unsigned long	0x00000001038e0828	0x00000001038e0828
rdi	unsigned long	0x000000000000000c	0x000000000000000c
rsi	unsigned long	0x00000001038e0598	0x00000001038e0598
rbp	unsigned long	0x00007000003224d0	0x00007000003224d0
rsp	unsigned long	0x0000700000322468	0x0000700000322468
```

**Comment by Vogtinator** [comment]
Fixed and v0.20 release updated!

**Comment by Candunc** [comment]
Can confirm that the fix worked, thanks for your help!
